flink-runtime/src/test/java/org/apache/flink/runtime/io/network/partition/consumer/InputBuffersMetricsTest.java

```diff
@@ -0,0 +1,180 @@
+
+/**
+ * Test for input buffer usage related metrics.
+ */
+public class InputBuffersMetricsTest {
+
+	@Test
+	public void testBufferUsageMetrics() throws IOException, InterruptedException {
+
+		CloseableRegistry closeableRegistry = new CloseableRegistry();
+
+		final NettyShuffleEnvironment network = new NettyShuffleEnvironmentBuilder().setIsCreditBased(true).build();
+
+		final SingleInputGate inputGate1 = new SingleInputGateBuilder()
+			.setNumberOfChannels(2)
+			.setIsCreditBased(true)
+			.setResultPartitionType(ResultPartitionType.PIPELINED_BOUNDED)
+			.setupBufferPoolFactory(network)
+			.build();
+
+		final SingleInputGate inputGate2 =  new SingleInputGateBuilder()
+			.setNumberOfChannels(2)
+			.setIsCreditBased(true)
+			.setResultPartitionType(ResultPartitionType.PIPELINED_BOUNDED)
+			.setupBufferPoolFactory(network)
+			.build();
+
+		int buffersPerChannel = 2;
+		int extraNetworkBuffersPerGate = 8;
+
+		try {
+			final ResultPartitionID resultPartitionId1 = new ResultPartitionID();
+			final ResultPartitionID resultPartitionId2 = new ResultPartitionID();
+
+			final ConnectionID connectionId1 = new ConnectionID(new InetSocketAddress("localhost", 5000), 0);
+			final ConnectionID connectionId2 = new ConnectionID(new InetSocketAddress("localhost", 5000), 0);
+
+			RemoteInputChannel remoteInputChannel1 = createUnknownInputChannel(network, inputGate1, resultPartitionId1, 0).toRemoteInputChannel(connectionId1);
+			inputGate1.setInputChannel(resultPartitionId1.getPartitionId(), remoteInputChannel1);
+
+			RemoteInputChannel remoteInputChannel2 = createUnknownInputChannel(network, inputGate1, resultPartitionId2, 1).toRemoteInputChannel(connectionId2);
+			inputGate1.setInputChannel(resultPartitionId2.getPartitionId(), remoteInputChannel2);
+
+			RemoteInputChannel remoteInputChannel3 = createUnknownInputChannel(network, inputGate2, resultPartitionId1, 0).toRemoteInputChannel(connectionId1);
+			inputGate2.setInputChannel(resultPartitionId1.getPartitionId(), remoteInputChannel3);
+
+			LocalInputChannel localInputChannel1 = createUnknownInputChannel(network, inputGate2, resultPartitionId2, 1).toLocalInputChannel();
+			inputGate2.setInputChannel(resultPartitionId2.getPartitionId(), localInputChannel1);
+
+			MetricGroup inputMetric = UnregisteredMetricGroups.createUnregisteredTaskMetricGroup().getIOMetricGroup().addGroup("buffers");
+
+			inputGate1.setup();
+			inputGate2.setup();
+
+			SingleInputGate[] inputGates = new SingleInputGate[]{inputGate1, inputGate2};
+			FloatingBuffersUsageGauge floatingBuffersUsageGauge = new FloatingBuffersUsageGauge(inputGates);
+			ExclusiveBuffersUsageGauge exclusiveBuffersUsageGauge = new ExclusiveBuffersUsageGauge(inputGates);
+			CreditBasedInputBufferPoolUsageGauge inputBufferPoolUsageGauge = new CreditBasedInputBufferPoolUsageGauge(floatingBuffersUsageGauge, exclusiveBuffersUsageGauge, inputGates);
+
+			inputMetric.gauge("floatingBuffersUsage", floatingBuffersUsageGauge);
```

Comment:
`inputMetric.gauge` seems not unnecessary, also the `inputMetrics` definition also not necessary.